test/unit/pages/signup.test.js

```diff
@@ -38,46 +38,61 @@ describe('Signup', function () {
       expect(console.warn.callCount).to.equal(0);
     });
 
-    it('should render waitlist when no key is set', function () {
+    it('should render signup-form when no key is set but checkInviteKey returns true', function () {
       console.warn = sinon.stub();
       var props = {
         onSubmit: sinon.stub(),
         onSubmitSuccess: sinon.stub(),
-        checkInviteKey: function(x, cb) { cb(true); },
+        checkInviteKey: function(x, cb) { return cb(true); },
         trackMetric: sinon.stub(),
         inviteKey: ''
       };
       var elem = React.createElement(Signup, props);
       var render = TestUtils.renderIntoDocument(elem);
-      var refreshButton = TestUtils.findRenderedDOMComponentWithClass(render, 'signup-form');
+      var signupForm = TestUtils.findRenderedDOMComponentWithClass(render, 'signup-form');
     });
 
-    it('should render signup form when key is set but is not valid', function () {
+    it('should render waitlist form when key is set but is not valid', function () {
       console.warn = sinon.stub();
       var props = {
         onSubmit: sinon.stub(),
         onSubmitSuccess: sinon.stub(),
-        checkInviteKey: function(x, cb) { cb(false); },
+        checkInviteKey: function(x, cb) { return cb(false); },
         trackMetric: sinon.stub(),
-        inviteKey: 'foobar'
+        inviteKey: 'wrong-key'
       };
       var elem = React.createElement(Signup, props);
       var render = TestUtils.renderIntoDocument(elem);
-      var refreshButton = TestUtils.findRenderedDOMComponentWithClass(render, 'waitlist');
+      var waitlist = TestUtils.findRenderedDOMComponentWithClass(render, 'waitlist');
     });
 
-    it('should render signup form when key is set and validates', function () {
+    it('should render signup-form when key is set and validates', function () {
       console.warn = sinon.stub();
       var props = {
         onSubmit: sinon.stub(),
         onSubmitSuccess: sinon.stub(),
-        checkInviteKey: function(x, cb) { cb(true); },
+        checkInviteKey: function(x, cb) { return cb(true); },
         trackMetric: sinon.stub(),
         inviteKey: 'foobar'
       };
       var elem = React.createElement(Signup, props);
       var render = TestUtils.renderIntoDocument(elem);
-      var refreshButton = TestUtils.findRenderedDOMComponentWithClass(render, 'signup-form');
+      var signupForm = TestUtils.findRenderedDOMComponentWithClass(render, 'signup-form');
+    });
+
+    it('should render signup-form when both key and email are set', function () {
```

Comment:
Just to check my understanding, I changed this test to have `checkInviteKey: function(x, cb) { return cb(false); }` and it passed because only the e-mail is actually required to render the sign-up. So, either add another test showing this case or maybe amend the language of this test to make it clear that when the e-mail is set, it doesn't matter if the key is?